lib/core/src/Cardano/Wallet.hs

```diff
@@ -595,6 +576,32 @@ restoreBlocks ctx wid blocks nodeTip = do
     db = ctx ^. dbLayer @s @t @k
     tr = ctx ^. logger
 
+    makeCheckpoint :: Wallet s t -> ExceptT ErrNoSuchWallet IO ()
+    makeCheckpoint cp = do
+        liftIO $ logInfo tr $
+            "Creating checkpoint at " <> pretty (currentTip cp)
+        DB.putCheckpoint db (PrimaryKey wid) cp
+
+    calculateMetadata
+        :: BlockchainParameters
+        -> BlockHeader
+        -> WalletMetadata
+        -> IO WalletMetadata
+    calculateMetadata bp h meta = do
+        -- NOTE: Safe because current time is after start time.
```

Comment:
Are we sure though?

Reply:
To some extend. Nodes can't start if the start time is after, so there would be no tip. But that's a weak guard I agree. 

Note that I didn't write that code :grimacing: ... I was just adding a comment to justify the unsafe pattern match here! If you see what I mean :no_mouth:

Reply:
> Note that I didn't write that code

Greater the reason 🙂 

In #788 I added a helper `syncProgressRelativeToCurrentTime`. If the current time is earlier than block0, we _could_ return `0%`. We could also fall-back to comparing against the node's tip. But maybe better to crash 🤷‍♂